runtime-modules/forum/src/lib.rs

```diff
@@ -790,6 +787,38 @@ decl_module! {
             Ok(())
         }
 
+        /// Update category
+        fn update_thread_archival_status(origin, actor: PrivilegedActor<T>, category_id: T::CategoryId, thread_id: T::ThreadId, new_archival_status: bool) -> dispatch::Result {
+            // Ensure data migration is done
+            Self::ensure_data_migration_done()?;
+
+            // Ensure actor can update category
+            let (category, thread) = Self::ensure_can_update_thread_archival_status(origin, &actor, &category_id, &thread_id)?;
+
+            if let Some(tmp_parent_category_id) = category.parent_category_id {
+                // Get path from parent to root of category tree.
+                let category_tree_path = Self::ensure_valid_category_and_build_category_tree_path(&tmp_parent_category_id)?;
```

Comment:
`ensure_valid_category_and_build_category_tree_path(&tmp_parent_category_id)?` is invoked as if it can fail, but it cannot, unless our own code has a bug. This is because `tmp_parent_category_id` does not come from the user, but from our own state, which we must presume is correct. Hence we really need a way to build the path you need without exposing an error which is not possible.

Reply:
This sub-issue is outdated. Check is done in a different way now.